File: examples/functions/rest-api-methods/products/_shared.go
package main

import "encoding/json"

type Product struct {
	ID    int     `json:"id"`
	Name  string  `json:"name"`
	Price float64 `json:"price"`
}

func catalogProducts() []Product {
	return []Product{
		{ID: 1, Name: "Widget", Price: 9.99},
		{ID: 2, Name: "Gadget", Price: 24.99},
	}
}

func jsonResponse(status int, payload interface{}) map[string]interface{} {
	body, _ := json.Marshal(payload)
	return map[string]interface{}{
		"status":  status,
		"headers": map[string]string{"Content-Type": "application/json"},
		"body":    string(body),
	}
}

File: examples/functions/rest-api-methods/products/_shared.lua
local cjson = require("cjson")

local M = {}

function M.catalog_products()
    return {
        { id = 1, name = "Widget", price = 9.99 },
        { id = 2, name = "Gadget", price = 24.99 },
    }
end

function M.json_response(status, payload)
    return {
        status = status,
        headers = { ["Content-Type"] = "application/json" },
        body = cjson.encode(payload),
    }
end

function M.trim_text(value)
    return tostring(value or ""):match("^%s*(.-)%s*$")
end

return M
